### PR TITLE
Add documentation yaml to collect documentation for internal DOCs too

### PR DESCRIPTION
**Pull request by wildcard**
Adding the documentation to internal repo
 similar to wix/wix-style-react#6425

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_569/

File: documentation.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  markdownOnlyAutoUpdate: true
+  docs:
+    - file: README.md
+      title: Getting Started
+    - file: docs/USAGE.md
+      title: Usage
+    - file: Contribution/docs/CONTRIBUTION.md
+      title: How to Contribute?
+    - file: docs/COMPONENT_GUIDELINES.md
+      title: Contribution/Guidelines for creating a `wix-ui-tpa` component
```
Comment (itaiche) [inline documentation.yaml:16]: Short titles, this is the left side menu:
`Contribution/Guidelines for creating a wix-ui-tpa component` => `Contribution/Guidelines`

File: documentation.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - file: docs/MIGRATING-V1-V2.md
+      title: Migrating `wix-ui-tpa` from v1 to v2
```
Comment (itaiche) [inline documentation.yaml:18]: `Migrating wix-ui-tpa from v1 to v2` => `Migration/V1 to V2`

File: documentation.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  status: ga
+  gitUrl: js
+  markdownOnlyAutoUpdate: true
+  docs:
+    - file: README.md
+      title: Getting Started
+    - file: docs/USAGE.md
+      title: Usage
+    - file: Contribution/docs/CONTRIBUTION.md
+      title: How to Contribute?
```
Comment (itaiche) [inline documentation.yaml:14]: `How to Contribute?` => `Contribution/How To Contribute`

File: documentation.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - file: docs/COMPONENT_GUIDELINES.md
+      title: Contribution/Guidelines for creating a `wix-ui-tpa` component
+    - file: docs/MIGRATING-V1-V2.md
+      title: Migrating `wix-ui-tpa` from v1 to v2
+    - file: docs/RTL.md
+      title: RTL Support
+    - file: docs/internal/VERSION_RELEASE.md
+      title: Version Release Instructions
+    - file: CHANGELOG.md
+      title: Change Log
```
Comment (itaiche) [inline documentation.yaml:24]: `Change Log` => `Change Log/V2`

File: documentation.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    - file: CHANGELOG-v1.md
+      title: Change Log v1
```
Comment (itaiche) [inline documentation.yaml:26]: `Change Log v1` => `Change Log/V1`

**Review by itaiche — CHANGES_REQUESTED** [review]
Minor changes.
Remember this represents the left side menu - it should be succinct

**Review by wildcard — COMMENTED** [review]
@itaiche Thanks for reviewing my PR, I've created commit suggestions, so you can commit to naming changes in your name 🍺

**Review by itaiche — APPROVED** [review]

**Comment by wildcard** [comment]
@jonathanadler Bro could you help us here ❤️

**Comment by jonathanadler** [comment]
> @jonathanadler Bro could you help us here ❤️

Awesome!
What does this do? 😁 
And how can I help?

**Comment by wildcard** [comment]
@jonathanadler 
This change will expose the project documentation on WixDocs internal.
we would like to merge & it's blocked even though @itaiche approved 🙈

**Comment by jonathanadler** [comment]
> @jonathanadler
> This change will expose the project documentation on WixDocs internal.
> we would like to merge & it's blocked even though @itaiche approved 🙈

I see 🤔 
So I think we can remove all the component Readme's, because we don't use them anymore, and we already removed the component README files from the generator

File: documentation.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    - file: src/components/NewCard/README.md
+      title: components/NewCard
+    - file: src/components/Counter/README.md
+      title: components/Counter
+    - file: src/components/IconToggle/README.md
+      title: components/IconToggle
+    - file: src/components/FloatingDropdown/README.md
+      title: components/FloatingDropdown
+    - file: src/components/Grid/README.md
+      title: components/Grid
```
Comment (wildcard) [inline documentation.yaml:66]: @jonathanadler I'm removing it, but do you have an alternative for those docs?

Funny thing about this list, I've use a shell command to generate it 🙈 

```sh
find src/components -name "*.md" | jq -Rn '[inputs | { file: ., title: . | sub("src/" ; "") | sub("/README.md";"") }]' | yq -y
```

```suggestion
```

**Comment by wildcard** [comment]
@jonathanadler I've removed them from docs.yml, but I guess you can delete the files too?

**Review by jonathanadler — APPROVED** [review]
Approved 💪 

I can merge, but if you want to merge please use the conventional commit as defined [here](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)

**Comment by jonathanadler** [comment]
Btw, is it possible to generate readme's in build time and publish to WixDocs?
Also, I'm assuming it shouldn't be a problem to change the titles, filenames, etc., right?

**Comment by wildcard** [comment]
No prob you can change the file as you will. 
@jonathanadler do you mean squash & merge using something like: "docs: add documentation yaml for internal docs". 

Merge is still blocked on my side

File: documentation.yaml (hunk 6 above)
Comment (jonathanadler) [reply documentation.yaml]: we use storybook for documentation (all the `*.index.story.tsx` files), which have examples and stuff like that.
I wonder if there's a way to integrate them with the internal docs.

Comment (shlomitc) [reply documentation.yaml]: No need to publish the components documentation. for this, we have the storybook which provides good documentation for components. WixDocs will not compete with that, so it should get only the standard MD files

Comment (wildcard) [reply documentation.yaml]: Thanks for explaining 😄 
So I guess we should take it with WixDocs to have an integration with Storybook 🙏 

cc: @alon-cohen

Comment (alon-cohen) [reply documentation.yaml]: I agree with the guys, we are not trying to solve what the storybook gives when speaking about components.

However, we should think about the flows that engineers might look for that information inside Wix Docs.
Maybe we can have 1 readme giving general info about the use-cases and a link to storybook?

WDYT? @shlomitc @wildcard @jonathanadler @itaiche

Comment (shlomitc) [reply documentation.yaml]: @alon-cohen This should be reflected in the main `README.md` file

Comment (wildcard) [reply documentation.yaml]: Sounds good! So the [README](https://github.com/wix/wix-ui-tpa/blob/master/README.md) should have a clear link to Storybook & explaining that on storybook you will find the entire catalog of the library's components.

> IMHO I still think WixDocs could parse the Storybook & create an index of everything (One place to search all components)

**Comment by itaiche** [comment]
> No prob you can change the file as you will.
> @jonathanadler do you mean squash & merge using something like: "docs: add documentation yaml for internal docs".
> 
> Merge is still blocked on my side

Feel free to join @alon-cohen's team and contribute this feature after design ;)

**Pull request merged**